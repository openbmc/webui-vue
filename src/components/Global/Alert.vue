<template>
  <b-alert :show="show" :variant="variant" :class="{ small }">
    <div
      v-if="
        variant == 'info' ||
          variant == 'success' ||
          variant == 'warning' ||
          variant == 'danger'
      "
      class="alert-icon"
    >
      <status-icon :status="variant" />
    </div>
    <div class="alert-content">
      <div class="alert-msg">
        <slot />
      </div>
    </div>
    <div class="alert-action">
      <slot name="action"></slot>
    </div>
  </b-alert>
</template>

<script>
import StatusIcon from '@/components/Global/StatusIcon';
import { BAlert } from 'bootstrap-vue';

export default {
  name: 'Alert',
  components: {
    BAlert: BAlert,
    StatusIcon: StatusIcon
  },
  props: {
    show: {
      type: Boolean,
      default: true
    },
    variant: {
      type: String,
      default: ''
    },
    small: Boolean
  }
};
</script>
