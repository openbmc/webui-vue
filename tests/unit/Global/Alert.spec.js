import { mount, createLocalVue } from '@vue/test-utils';
import Alert from '@/components/Global/Alert';

const localVue = createLocalVue();

describe('Alert.vue', () => {
  const wrapper = mount(Alert, {
    localVue,
    propsData: {
      show: true,
      variant: 'Test a type of alert',
      small: false,
    },
  });
  it('should exist', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('should show alert', async () => {
    await wrapper.setProps({ show: true });
    expect(wrapper.find('.alert').exists()).toBe(true);
  });
  it('should hide alert', async () => {
    await wrapper.setProps({ show: false });
    expect(wrapper.find('.alert').exists()).toBe(false);
  });
  it('should render correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
