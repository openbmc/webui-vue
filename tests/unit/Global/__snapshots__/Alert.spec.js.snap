// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert.vue should render correctly 1`] = `
<transition-stub
  class=""
  css="true"
  enteractiveclass=""
  enterclass=""
  entertoclass="show"
  leaveactiveclass=""
  leaveclass="show"
  leavetoclass=""
  name=""
/>
`;
